doc/source/user/workbench.rst

```diff
@@ -0,0 +1,178 @@
+ONLY `executable`. Additionally, there are  application-specific inputs
+that can be provided to the plugins such as::
+
+    `extra_args` (multiple apps) : Extra arguments to applications
+    `plotfl_to_csv` (RELAP5) : Option to convert `plotfl` file to `csv` file
+    `conv_channel` (SAS) : Path to `CHANNELtoCSV.x`
+    `conv_primar4` (SAS) : Path to `PRIMAR4toCSV.x`
+    `auto_link_files` (Dakota) : List of files for Dakota to link automatically
+    `scores` (OpenMC) : List of filters for tallies
+    `score_names` (OpenMC) : List of user-given names for tally filters
+
+Workflow level
+~~~~~~~~~~~~~~
+
+The `workflow_level1` block is required. The plugin to be used is specified
+by the `plugin` keyword::
+
+    plugin = ID1
+
+where 'ID1' is the ID of the plugin provided in the `plugins` block. The
+`variable` sub-block is where the values of the templated variables are
+specified, as shown below::
+
+    variables{
+        param(T_inlet) {value = 873.15}
+        param(T_outlet) {value = 1000.15}
+        param(flow_area) {value = 3.14e-4}
+        param(component_name) {value = pipe1}
+        param(Dh) {value = 1 unit = m}
+        param(location) {list = [top bottom left right]}
+        param(length) {list = [3.0 5.0 7.0 10.0]}
+        param(T_mean) {func = "0.5 * ( T_inlet + T_outlet )"}
+        param(wall_bc) {bool = "True"}
+    }
+
+Each `param` has a unique ID represented by the string in the parantheses
+shown in the snippet above. A `param` can accept different types of inputs
+depending on the provided key word provided. The `value` keyword is used
+for when a numeric or a string is provided. A `unit` keyword can be added
+if a user intends to utilize WATTS' internal unit-conversion capability.
+The `list` keyword is used for a list of numerics or strings. The `bool`
+keyword is used for booleans. The `func` keyword is used when a user
+intends to perform arithmetic with existing `param`. Note that each elemet
+in `func` must be separated by a space.
+
+Execution
++++++++++
+
+Three types of executions can be performed by WATTS on Workbench, namely
+direct execution, parametric study, and Picard iteration.
+
+Direct execution
+~~~~~~~~~~~~~~~~
+
+Direct execution is the simplest execution type. The user only needs to
+provide `workflow_dir`, `plugins` block, and `workflow_level1` block to
+perform direct execution.
+
+Parametric study
+~~~~~~~~~~~~~~~~
+
+To perform parametric study, a `parametric` block needs to be added to
+the `workflow_level1` block as follows::
+
+    parametric{
+        changing_params = "heat_source"
+        changing_values = [0, 1e5, 2e5, 3e5]
+    }
+
+where `changing_params` is the parameter whose values are varied and
+`changing_values` is a list of intended varied values.
+
+Picard iteration
+~~~~~~~~~~~~~~~~
+
+To perform Picard iteration, the `iteration` block needs to be added
+to the `workflow_level1` block::
+
+    iteration{
+        plugin_main = ID1
```

Comment:
not sure why plugin_main and plugin_sub needs specifying - I would have guessed we would use the plugin specified under workflow_level1. I see why this is needed currently because we are specifying how data is transferred from 1 app to the other. However, could this be done directly at the workflow/plugin/params level so that it can generalized in case we have additional apps involved in the workflow?

Reply:
No longer need to specify `plugin_main` but still need to specify `plugin = ID2`. This is specifically for `iteration` only because of how the iteration workflow is set up currently. I think it's less confusing this way but we can change it if needed.